opm/autodiff/Compat.hpp

```diff
@@ -21,229 +21,41 @@
+#include <opm/core/props/BlackoilPhases.hpp>
+#include <vector>
 
 namespace Opm {
 
-inline std::vector< double > destripe( const std::vector< double >& v,
-                                       size_t stride,
-                                       size_t offset ) {
-
-    std::vector< double > dst( v.size() / stride );
-
-    size_t di = 0;
-    for( size_t i = offset; i < v.size(); i += stride ) {
-        dst[ di++ ] = v[ i ];
-    }
-
-    return dst;
-}
-
-
-
-
-
-inline std::vector< double >& stripe( const std::vector< double >& v,
-                                      size_t stride,
-                                      size_t offset,
-                                      std::vector< double >& dst ) {
-
-    /* does little range checking etc; for future revisions */
-    size_t vi = 0;
-    for( size_t i = offset; i < dst.size(); i += stride ) {
-        dst[ i ] = v[ vi++ ];
-    }
-
-    return dst;
-}
-
-
-
-
-
-
-
-    inline data::Solution simToSolution( const SimulationDataContainer& reservoir,
-                                         PhaseUsage phases ) {
-    data::Solution sol;
-    sol.insert( "PRESSURE", UnitSystem::measure::pressure, reservoir.pressure() , data::TargetType::RESTART_SOLUTION);
-    sol.insert( "TEMP"    , UnitSystem::measure::temperature, reservoir.temperature() , data::TargetType::RESTART_SOLUTION );
-
-    const auto ph = reservoir.numPhases();
-    const auto& sat = reservoir.saturation();
-
-    const auto aqua = BlackoilPhases::Aqua;
-    const auto vapour = BlackoilPhases::Vapour;
-
-    if( phases.phase_used[ aqua ] ) {
-        sol.insert( "SWAT", UnitSystem::measure::identity, destripe( sat, ph, phases.phase_pos[ aqua ] ) , data::TargetType::RESTART_SOLUTION );
-    }
-
-    if( phases.phase_used[ vapour ] ) {
-        sol.insert( "SGAS", UnitSystem::measure::identity, destripe( sat, ph, phases.phase_pos[ vapour ] ) , data::TargetType::RESTART_SOLUTION );
-    }
-
-    if( reservoir.hasCellData( BlackoilState::GASOILRATIO ) ) {
-        sol.insert( "RS", UnitSystem::measure::gas_oil_ratio, reservoir.getCellData( BlackoilState::GASOILRATIO ) , data::TargetType::RESTART_SOLUTION );
-    }
-
-    if( reservoir.hasCellData( BlackoilState::RV ) ) {
-        sol.insert( "RV", UnitSystem::measure::oil_gas_ratio, reservoir.getCellData( BlackoilState::RV ) , data::TargetType::RESTART_SOLUTION );
-    }
-
-    if (reservoir.hasCellData( BlackoilSolventState::SSOL)) {
-        sol.insert( "SSOL", UnitSystem::measure::identity, reservoir.getCellData( BlackoilSolventState::SSOL ) , data::TargetType::RESTART_SOLUTION );
-    }
-
-    return sol;
-}
-
-
-
-
-
+    // Forward declarations
+    class SimulationDataContainer;
+    class WellStateFullyImplicitBlackoil;
+    class WellStateFullyImplicitBlackoilDense;
 
+    std::vector< double > destripe( const std::vector< double >& v,
+                                    size_t stride,
+                                    size_t offset );
```

Comment:
Ah, sorry: I did not notice that this was a traditional header/source file combo.